client/app/app.js

```diff
@@ -0,0 +1,51 @@
+import { h, Component } from 'preact'
```

Comment:
Personally, I prefer to have a file extension that describes what is in the file. If I got a .js, file I am not expecting to find some JSX or other stuff inside.
So my preference goes to separated .js and .jsx file.
However I may be missing a point.